matchms/filtering/metadata_processing/repair_smiles_from_compound_name.py

```diff
@@ -10,8 +15,8 @@
 
 
 def repair_smiles_from_compound_name(spectrum_in: Spectrum,
-                                     annotated_compound_names_file,
-                                     mass_tolerance=0.1):
+                                     annotated_compound_names_file: Optional[str] = None,
+                                     mass_tolerance: float = 0.1):
     """Adds annotations (smiles, inchi, inchikey) based on compound name
 
     Based on a table of compound names and smiles matches (stored in a csv file) this function
```

Comment:
The docstring is outdated/not correct. It should mention more clearly that it also contains a PubChem library search.

That being said, it might also be better to make PubChem more prominent here. Maybe `_get_pubchem_compound_name_annotation` or similar. Plus, we might (now or in the future) also add a flag to specify different databases?

Reply:
Helge also mentioned that they already have a package that has that kind of functionality. Since it builds on matchms it was not really easy to build directly into pipeline. Therefore I did in the end still build the pubchem lookup in here, instead of using their package. But it might be better to later use his package for this, to prevent implementing the same thing twice. 

I will update the docstring.